File: src/main/java/com/github/trqhxrd/untitledgame/engine/Display.java
package com.github.trqhxrd.untitledgame.engine;

import lombok.Getter;

public class Display {
    @Getter
    private final int height = 1080;
    @Getter
    private final int width = 1920;
    @Getter
    private final String title = "DER NAME";

    private static Display display;

    public static Display get() {
        if (Display.display == null) Display.display = new Display();
        return display;
    }
}

File: src/main/kotlin/com/github/trqhxrd/untitledgame/engine/gui/display/Display.kt
package com.github.trqhxrd.untitledgame.engine.gui.display

import java.awt.Graphics
import java.awt.Graphics2D
import javax.swing.JComponent

class Display : JComponent() {

    val elements = mutableListOf<DisplayElement>()

    override fun paintComponent(g: Graphics) {
        this.elements.forEach { it.draw(g as Graphics2D) }
    }
}
